src/ompl/tools/thunder/src/SPARSdb.cpp

```diff
@@ -1278,13 +1285,14 @@ void ompl::geometric::SPARSdb::resetFailures()
 }
 
 void ompl::geometric::SPARSdb::findGraphNeighbors(base::State *st, std::vector<Vertex> &graphNeighborhood,
-                                                  std::vector<Vertex> &visibleNeighborhood)
+                                                  std::vector<Vertex> &visibleNeighborhood, double radius)
 {
     visibleNeighborhood.clear();
     stateProperty_[queryVertex_] = st;
-    nn_->nearestR(queryVertex_, sparseDelta_, graphNeighborhood);
+    if (radius < 0) radius = sparseDelta_;
+    nn_->nearestR(queryVertex_, radius, graphNeighborhood);
     if (verbose_ && false)
-        OMPL_INFORM("Finding nearest nodes in NN tree within radius %f", sparseDelta_);
+        OMPL_INFORM("Finding nearest nodes in NN tree within radius %f", radius);
```

Comment:
what's the difference between `radius` and `sparseDelta_`? is this a bug upstream?

Reply:
This is part of the change I did to the algorithm. all calls to findGraphNeighbors used SparseDelta as their radius, I added this parameter to have control over it. I use different radii in different places I call the function.

Reply:
then you should give it a default value of `sparseDelta_`

Reply:
I think I tried that and it didnt work so this was my way of getting around that. I will try again!

Reply:
/src/ompl/src/ompl/tools/thunder/SPARSdb.h:619:119: error: invalid use of non-static data member ‘ompl::geometric::SPARSdb::sparseDelta_’
  619 |             bool findGraphNeighbors(const base::State *state, std::vector<Vertex> &graphNeighborhood, double radius = sparseDelta_);
  
  
  it gives me this error when I try to default radius = sparseDelta_ in function declaration

Reply:
set it inside the function

Reply:
what do you mean by optional input? design two functions with one that accepts it and one that does not?

Reply:
https://dexairobotics.atlassian.net/wiki/spaces/DEX/pages/913211430/Code+Guidelines#Functions